packages/gatsby-source-filesystem/src/utils.js

```diff
@@ -13,3 +13,16 @@ const Url = require(`url`)
 export function getRemoteFileExtension(url) {
   return path.parse(Url.parse(url).pathname).ext
 }
+
+/**
+ * getRemoteFileName
+ * --
+ * Parses remote url to retrieve remote file name
+ *
+ *
+ * @param  {String}          url
+ * @return {String}          filename
+ */
+export function getRemoteFileName(url) {
+  return path.parse(Url.parse(url).pathname).name
```

Comment:
Since this block is pretty much directly shared between `getRemoteFileExtension` and this function, maybe encapsulate the following code into a shared function so as to not duplicate work--and later, if we need to tweak, it'll be an easy swap in _one_ place rather than a few, e.g.

```js
function getParsedPath(url) {
  return path.parse(Url.parse(url).pathname)
}

export function getRemoteFileExtension(url) {
  return getParsedPath(url).ext
}

export function getRemoteFileName(url) {
  return getParsedPath(url).name
}
```

Reply:
Fixed. Thanks for the review.